### PR TITLE
make the UI module based

### PR DESCRIPTION
**Pull request by alicebob**
Not finished, but:

This changes the UI .js file to be generated by browserify. The individual .js files are now proper modules. Unfortunately, the way the code refers to other modules is over the `ui` global. Ideally all of those would be replaced by proper requires, for example, instead of using `ui.toolarea` it should be `var toolarea = require('./ToolArea.js')`. That's a change which can be done after this. So for now the new `core.js` re-assembles things as they already were.

The browserify setup is completely vanilla, and there is no uglify or anything. The idea would be to plug in https://github.com/TypeStrong/tsify for Typescript.

Currently the UI .js file is generated by `make ui`. There is also https://www.npmjs.com/package/grunt-browserify to help. Who knows. Didn't look yet.

### CODE REVIEW

**Comment by alicebob** [comment]
browserify help: https://github.com/browserify/browserify-handbook

**Comment by robx** [comment]
Thanks, this looks promising!

File: src-ui/js/core.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+var boot = require('./ui/Boot.js');
+
+function load() {
+	if (boot.importData()) {
+		boot.startPuzzle();
+	} else {
+		setTimeout(load, 0);
+	}
+};
+window.onload = load;
```
[the remark this answers is not in the document]
Comment (alicebob) [reply src-ui/js/core.js]: ok, back to what it kinda was

**Comment by alicebob** [comment]
I changed main pzpr to be modules based. The way this currently works is that every variant is `required()` (aka: every variant is embedded in the main .js file). I only did sliter and nurikabe.

build command for now: `make ui pzpr`.

Not convinced this all is really useful :(

(Somewhere along the line the browser language stopped being detected anymore, but that looks minor.)

**Comment by alicebob** [comment]
Will come back if this ever gets nice enough.

**Pull request closed without merging**